docs/site/Parsing-requests.md

```diff
@@ -0,0 +1,202 @@
+- Performs validation on the parameters and body data.
+
+### Parsing Raw Data
+
+The code below defines a typical endpoint by decorating a controller method with
+rest decorators.
+
+```ts
+class TodoController {
+  constructor(@repository(TodoRepository) protected todoRepo: TodoRepository) {}
+
+  @put('/todos/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todo: Todo,
+  ): Promise<boolean> {
+    return await this.todoRepo.replaceById(id, todo);
+  }
+}
+```
+
+An operation specification will be generated in-memory to describe it, and raw
+data are parsed from request according to the specification. In the example
+above, the first parameter is from source `path`, then its value will be parsed
+from a request's path.
+
+{% include note.html title="Controller documentation" content="See [controller](Controller.md) for more details of defining an endpoint." %}
+
+{% include note.html title="OpenAPI operation object" content="See
+[OpenAPI operation object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#operationObject)
+to know more about its structure." %}
+
+### Coercion
+
+The parameters parsed from path, header, and query of a http request are always
+in the string format when using the `http` module in Node.js to handle requests.
+Therefore when invoking a controller function, a parameter need to be converted
+to its corresponding JavaScript runtime type, which is inferred from its
+parameter specification.
+
+Take the example operation `replaceTodo` in section
+[parsing raw data](#parsing-raw-data) which takes in a number `id` as the first
+input. Without coercion,`id` would have to be manually cast into the number type
+before it can be used as seen below:
+
+```ts
+@put('/todos/{id}')
+async replaceTodo(
+  @param.path.number('id') id: number,
+  @requestBody() todo: Todo,
+): Promise<boolean> {
+  // NO need to do the "string to number" convertion now,
+  // coercion automatically handles it for you.
+  id = +id;
+  return await this.todoRepo.replaceById(id, todo);
+}
+```
+
+### Validation
+
+Validations are applied on the parameters and the request body data. They also
+use OpenAPI specification as the reference to infer the validation rules.
+
+#### Parameters
+
+We have the data type safety check for the parameters parsed from header, path,
+and query. For example, if a parameter should be an integer, then a number with
+decimal like "1.23" would be rejected.
+
+You can specify a parameter's type by calling shortcut decorators of `@param`
+like `@param.query.integer()`. A list of available shortcuts can be found in the
+[api documents](https://apidocs.strongloop.com/@loopback%2fdocs/openapi-v3.html#param).
+Check out the section on
+[parameter decorators](Decorators.md#parameter-decorator) for instructions on
+how to decorate the controller parameter.
+
+Here are our default validation rules for each type:
+
+- number: validated by `isNaN(Number(data))`.
```

Comment:
nitpick: I suggest to either remove `.` for each line of the list or add it to be consistent

Reply:
@bajtos hmm...I added it for each line....on which line you didn't find it?